code-challenges.test.js

```diff
@@ -15,42 +15,131 @@
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const fibLength1 = 6
-// Expected output: [1, 1, 2, 3, 5, 8]
+//input: fibLength1
+//output: Expected output: [1, 1, 2, 3, 5, 8]
 
-const fibLength2 = 10
-// Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+// input: fibLength2
+// output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+describe("fibArray", () => {
+    const fibLength1 = 6
+    // Expected output: [1, 1, 2, 3, 5, 8]
+    const fibLength2 = 10
+    // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+    it("Takes in a  number and returns  that length  containing fibinacci sequence in a new array", () => {
+        expect(fibArray(fibLength1)).toEqual([1, 1, 2, 3, 5, 8])
+        expect(fibArray(fibLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+    })
+})
+
+// ReferenceError: fibArray is not defined
 
 // b) Create the function that makes the test pass.
 
+// Pseudo ---- >
+// Create a function named fibArray
+// Pass in a num paramater that will hold a number value when passed in
+// Create a new array using the Array object.
+// Use the from static method to set the iteration length tho the size of the number being passed
+// Use the reduce method to call a call back function on each iteration of the array storing the value from th calculation
+// As the function runs concat each iterations return value into a new array.
+// use a conditional statement index is greater than one
+// use ternary operators to seperate true and false results stored in the array
+// If index is greater then 1 then take the addition of the last to values and store that value.
+// Outside of testing console log to check the result in the terminal
+
+const fibArray = (num) => Array.from({length: num}).reduce((accumulator,_,i) => accumulator.concat(i  > 1 ? accumulator[i - 1] + accumulator[i - 2]: 1),
```

Comment:
Nice!  Really cool approach that I have never seen before, and really thorough pseudocode that breaks it all down very clearly.  Great work!